### PR TITLE
cleanup(tests,pausing): Use client instead of virtctl

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

Use the KubeVirt client instead of using virtctl through a NewRepeatableVirtctlCommand where applicable in
tests/migration/migration.go.

Internally virtctl uses the same client call, so using virtctl does not provide any additional value.

This also drops duplicate tests combinations which can only occur when
using virtctl (e.g. pausing VM twice and pausing VMI twice, the client can
only pause VMIs and internally virtctl only gets a VM before pausing
the corresponding VMI) and reduces the amount of tests by combining the
test for VMI Ready and Paused condition and by dropping the test pausing
and unpausing a VMI multiple times, becomes this is already covered by the
check-tests-for-flakes lane.

The correctness of API calls made by `virtctl pause` and `virtctl unpause` is already covered in unit tests specified in `pkg/virtctl/pause/pause_test.go`.

The goal of this and further PRs is to reduce the amount of unnecessary uses of
virtctl in the e2e tests to stabilize tests, to reduce the number of flakes and to
remove redudant tests.

Before this PR:

NewRepeatableVirtctlCommand is used unnecessarily.

After this PR:

The KubeVirt client is used instead.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/pausing_test.go [hunk 1]
```diff
@@ -53,136 +52,98 @@ import (
...
+			Eventually(matcher.ThisVMI(vmi), 30*time.Second, time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstancePaused))
 
-				err = virtClient.VirtualMachineInstance(vmi.Namespace).Pause(context.Background(), vmi.Name, &v1.PauseOptions{})
+			By("Unpausing VMI")
+			err = virtClient.VirtualMachineInstance(vmi.Namespace).Unpause(context.Background(), vmi.Name, &v1.UnpauseOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(matcher.ThisVMI(vmi), 30*time.Second, time.Second).Should(matcher.HaveConditionMissingOrFalse(v1.VirtualMachineInstancePaused))
+		})
+
+		It("[test_id:3225]should signal paused state with condition when paused multiple times", func() {
```
Comment (dankenigsberg) [inline tests/pausing_test.go:82]: Do you know why we need to start/stop multiple times?

What is the value of `test_id:4597` if we repeat it three times in `test_id:3225`?

File: tests/pausing_test.go [hunk 2]
```diff
@@ -53,136 +52,98 @@ import (
 )
 
 var _ = Describe("[rfe_id:3064][crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-compute]Pausing", decorators.SigCompute, func() {
+	const cirrosStartupTimeout = 90
```
Comment (dankenigsberg) [inline tests/pausing_test.go:55]: nit, and not introduced by this PR, but this const is better set closer to its single user.

File: tests/pausing_test.go [hunk 3]
```diff
@@ -53,136 +52,98 @@ import (
...
+			Expect(err).ToNot(HaveOccurred())
 
-				By("Checking that VMI remains paused")
-				Consistently(matcher.ThisVMI(vmi), 5*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstancePaused))
-			})
+			By("Checking that VMI remains paused")
+			Consistently(matcher.ThisVMI(vmi), 5*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstancePaused))
 		})
 
 		It("should not appear as ready when paused", func() {
```
Comment (dankenigsberg) [inline tests/pausing_test.go:142]: How about adding this expectation to another test?
Starting a VM and pausing it once for every condition seems awfully expensive.

**Review by dankenigsberg — CHANGES_REQUESTED** [review]
Thanks for removing seemingly redundant tests. We should not test the very same thing twice. I left few comments about the remaining tests, but they are not strictly related to `virtctl`.

Would you explain in the PR text where the `virtctl pause` is tested? I think we should still ensure that it calls the expected subresource, e.g in a unit test or an API test.

File: tests/pausing_test.go (hunk 2 above)
Comment (0xFelix) [reply tests/pausing_test.go]: Updated

File: tests/pausing_test.go (hunk 3 above)
Comment (0xFelix) [reply tests/pausing_test.go]: Good suggestion, addressed!

File: tests/pausing_test.go (hunk 1 above)
Comment (dankenigsberg) [reply tests/pausing_test.go]: > Probably to rule out any flakes.

I dont' understand you: repeating a raceful test three times would only increase the probability of the race occurring.

> I was thinking about dropping it. Would you agree?

I don't know. Luckily we have @stu-gott around, maybe he remembers. In any case, I'd do it only in a separate commit, with a very clear commit message.

Comment (0xFelix) [reply tests/pausing_test.go]: > I dont' understand you: repeating a raceful test three times would only increase the probability of the race occurring.

Exactly, the test is run three times in succession to ensure it passes every time. That is what the `check-tests-for-flakes` lane does too. All modified tests are run five times to ensure they pass every time. Please correct me if I'm wrong.

Comment (dankenigsberg) [reply tests/pausing_test.go]: I don't get why pause/unpause is different than any other test that it needs this repetition.

**Pull request merged**